File: python/chat_completion.py
#!/usr/bin/env python3
"""
Python Example: Chat Completion with OpenAI-Compatible API

This script demonstrates how to make a simple chat completion request
using the official OpenAI Python client with a custom base URL.
"""

import os
import sys

# Load .env file if python-dotenv is available
try:
    from dotenv import load_dotenv
    load_dotenv()
except ImportError:
    pass

# Configuration from environment
API_KEY = os.getenv("API_KEY")
BASE_URL = os.getenv("BASE_URL", "https://tken.shop/v1")
MODEL = os.getenv("MODEL", "gpt-4o-mini")


def check_config():
    """Verify required configuration is present."""
    if not API_KEY:
        print("Error: API_KEY environment variable is not set")
        print("Set it with: export API_KEY=your_api_key_here")
        print("Or create a .env file based on .env.example")
        sys.exit(1)


def main():
    check_config()

    try:
        from openai import OpenAI
    except ImportError:
        print("Error: openai package not installed")
        print("Run: pip install -r requirements.txt")
        sys.exit(1)

    # Initialize client with custom base URL
    client = OpenAI(
        api_key=API_KEY,
        base_url=BASE_URL
    )

    print(f"API: {BASE_URL}")
    print(f"Model: {MODEL}")
    print("-" * 40)

    try:
        response = client.chat.completions.create(
            model=MODEL,
            messages=[
                {"role": "user", "content": "Say hello in one sentence."}
            ],
            max_tokens=50,
            temperature=0.7
        )

        print("Success!")
        print("-" * 40)
        print(f"Response: {response.choices[0].message.content}")
        print(f"Model: {response.model}")
        print(f"Tokens: {response.usage.total_tokens}")

    except Exception as e:
        print(f"Error: {e}")
        sys.exit(1)


if __name__ == "__main__":
    main()

File: node/chat_completion.js
#!/usr/bin/env node
/**
 * Node.js Example: Chat Completion with OpenAI-Compatible API
 *
 * This script demonstrates how to make a simple chat completion request
 * using the official OpenAI JavaScript client with a custom base URL.
 */

require('dotenv').config();

const API_KEY = process.env.API_KEY;
const BASE_URL = process.env.BASE_URL || 'https://tken.shop/v1';
const MODEL = process.env.MODEL || 'gpt-4o-mini';

function checkConfig() {
    if (!API_KEY) {
        console.error('Error: API_KEY environment variable is not set');
        console.error('Set it with: export API_KEY=your_api_key_here');
        console.error('Or create a .env file based on .env.example');
        process.exit(1);
    }
}

async function main() {
    checkConfig();

    const { OpenAI } = require('openai');

    const client = new OpenAI({
        apiKey: API_KEY,
        baseURL: BASE_URL
    });

    console.log(`API: ${BASE_URL}`);
    console.log(`Model: ${MODEL}`);
    console.log('-'.repeat(40));

    try {
        const response = await client.chat.completions.create({
            model: MODEL,
            messages: [
                { role: 'user', content: 'Say hello in one sentence.' }
            ],
            max_tokens: 50,
            temperature: 0.7
        });

        console.log('Success!');
        console.log('-'.repeat(40));
        console.log(`Response: ${response.choices[0].message.content}`);
        console.log(`Model: ${response.model}`);
        console.log(`Tokens: ${response.usage.total_tokens}`);

    } catch (error) {
        console.error(`Error: ${error.message}`);
        process.exit(1);
    }
}

main();
